### ISSUE TITLE
Incorrect "on hand" numbers

### ISSUE THREAD

**Issue by ehuelsmann** [ledgersmb/LedgerSMB#6430]
while testing #6428, I ran into incorrect OH numbers by running the following sequence of actions:

1. Create a new vendor
2. Create a part
3. Create a purchase order with the part
4. Receive the part from the purchase order
5. Create a vendor invoice from the purchase order

Result: The part purchase has been added to "On Hand" twice.

**Comment by tonka128** [comment]
Yup - we run into this all the time - we have had to created a cludge script to run on the database to correct stock on a regular basis
NOTE: also an issue with draft invoices that cant post (ie something that you have not enough sales of stock to reverse) every attemtp to post the draft with add another level of stock adjustment.

**Comment by ehuelsmann** [comment]
My research notes about inventory being off:

* There are 3 ways to adjust inventory in OE alone:

  - adj_onhand (seems to be unused/dead code?)
  - adj_inventory (used in OE::save, called from oe.pl::create_backorder)
  - save_inventory (used in oe.pl::done -- which is part of shipping)

* Inventory is adjusted when deleting an order (but not added to when creating one)

* Inventory is increased by the amount shipped when creating a backorder

* Inventory is increased by the amount invoiced when creating an invoice

* Shipped items on an order are removed on order deletion

* Shipping/Receiving does not modify the onhand amount
